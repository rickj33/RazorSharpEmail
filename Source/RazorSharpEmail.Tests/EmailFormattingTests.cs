using System.IO;
using ApprovalTests.Reporters;
using Email.Models;
using FluentAssertions;
using NUnit.Framework;

namespace RazorSharpEmail.Tests
{
    [TestFixture]
    [UseReporter(typeof(DiffReporter))]
    public class EmailFormattingTests
    {
        private IEmailGenerator _emailGenerator;

        [TestFixtureSetUp]
        public void TestFixtureSetUp()
        {
            _emailGenerator = Given.AnInitializedEmailFormatter();
        }

        [Test]
        public void Should_format_email_from_model()
        {
            var email = _emailGenerator.Generate(
                new Welcome
                {
                    FirstName = "Michael",
                    Message = "Hello World!",
                    Url = "http://google.com"
                });
            
            ApprovalTests.Approvals.Verify(email.Everything());
        }

        [Test]
        public void Should_inline_styles() {
            var email = _emailGenerator.Generate(new object(), "Simple");

            var premailer = new PreMailer.Net.PreMailer();
            email.HtmlBody = premailer.MoveCssInline(email.HtmlBody, false);

            ApprovalTests.Approvals.Verify(email.HtmlBody);
        }
    }
}