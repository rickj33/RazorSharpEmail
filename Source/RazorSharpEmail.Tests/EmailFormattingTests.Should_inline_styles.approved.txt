<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd"><html xmlns="http://www.w3.org/1999/xhtml"><head>
        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
        <title>Simple!</title>

        <style type="text/css">
            * {
	margin:0;
	padding:0;
}
* { font-family: sans-serif; }

a { color: #2BA6CB;}

.btn {
	color: #FFF;
	padding:10px 16px;
	margin-right:10px;
}
        </style>
    </head>
    <body style="font-family: sans-serif; margin: 0; padding: 0;">
        <a class="btn" style="font-family: sans-serif; margin: 0 10px 0 0; padding: 10px 16px; color: #2ba6cb;" href="http://microsoft.com">Simple</a>
    

</body></html>